chrysom/basicClient.go

```diff
@@ -73,17 +72,17 @@ type BasicClientConfig struct {
 
 	// Logger to be used by the client.
 	// (Optional). By default a no op logger will be used.
-	Logger log.Logger
+	Logger *zap.Logger
 }
 
 // BasicClient is the client used to make requests to Argus.
 type BasicClient struct {
 	client       *http.Client
 	auth         acquire.Acquirer
 	storeBaseURL string
-	logger       log.Logger
+	logger       *zap.Logger
 	bucket       string
-	getLogger    func(context.Context) log.Logger
+	getLogger    sallust.GetLoggerFunc
```

Comment:
We don't want to make a custom type withing `sallust` for this.  That creates a dependency on `sallust` for no other reason than for that type.  Using `func(context.Context) *zap.Logger` is actually better usage.